packages/nextcloud/bin/generate_exports.dart

```diff
@@ -0,0 +1,37 @@
+// ignore_for_file: depend_on_referenced_packages
+import 'dart:io';
+
+import 'package:dynamite/src/helpers/dart_helpers.dart';
+import 'package:path/path.dart' as p;
+
+void main() {
+  final files =
+      Directory('lib/src/api').listSync().cast<File>().where((final file) => file.path.endsWith('.openapi.dart'));
+
+  for (final file in files) {
+    final basename = p.basename(file.path);
+    final id = basename.substring(0, basename.length - 13);
+    final variablePrefix = toDartName(id);
+    final classPrefix = toDartName(id, uppercaseFirstCharacter: true);
+
+    final exports = ["export 'src/api/$id.openapi.dart';"];
+    if (File('lib/src/helpers/$id.dart').existsSync()) {
+      exports.add("export 'src/helpers/$id.dart';");
+    }
+
+    File('lib/$id.dart').writeAsStringSync('''
+import 'package:nextcloud/src/api/$id.openapi.dart';
+import 'package:nextcloud/src/client.dart';
+
+${exports.join('\n')}
+
+// ignore: public_member_api_docs
+extension ${classPrefix}Extension on NextcloudClient {
+  static final _$variablePrefix = Expando<${classPrefix}Client>();
```

Comment:
why exactly an expando?

Reply:
Extensions can't introduce new variables. Using a simple map would keep the reference to the object until the whole program exits, so we wouldn't get rid of unreachable objects. Expando does exactly that for us: https://api.dart.dev/stable/3.1.3/dart-core/Expando-class.html

Reply:
> Using a simple map would keep the reference to the object until the whole program exits, so we wouldn't get rid of unreachable objects.

Can you further explain this/ point me to some documentation

Reply:
So previously I had something like
```dart
static final _core = <NextcloudClient, CoreClient>{};
```
which worked just fine.
Now this static constant stays alive during the whole program lifecycle.
This means the values inside the map are never cleaned up by GC when the respective NextcloudClient becomes unreachable.
To correctly free the object you can use Expando since it will remove the value once it becomes unreachable.

Reply:
Ah now I get it.

Thanks.

Reply:
I also didn't know about Expando, but it's a pretty cool feature :)